go-controller/pkg/ovn/policy.go

```diff
@@ -1021,50 +855,37 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) {
 	for i, egressJSON := range policy.Spec.Egress {
 		klog.V(5).Infof("Network policy egress is %+v", egressJSON)
 
-		egress := newGressPolicy(knet.PolicyTypeEgress, i)
+		egress := newGressPolicy(knet.PolicyTypeEgress, i, policy.Namespace, policy.Name)
 
 		// Each egress rule can have multiple ports to which we allow traffic.
 		for _, portJSON := range egressJSON.Ports {
 			egress.addPortPolicy(&portJSON)
 		}
 
-		hashedLocalAddressSet := ""
-		// peerPodAddressMap represents the IP addresses of all the peer pods
-		// for this egress.
-		peerPodAddressMap := make(map[string]bool)
 		if hasAnyLabelSelector(egressJSON.To) {
-			// localPeerPods represents all the peer pods in the same
-			// namespace to which we need to allow traffic.
-			localPeerPods := fmt.Sprintf("%s.%s.%s.%d", policy.Namespace,
-				policy.Name, "egress", i)
-
-			hashedLocalAddressSet = hashedAddressSet(localPeerPods)
-			createAddressSet(localPeerPods, hashedLocalAddressSet, nil)
-			egress.addAddressSet(hashedLocalAddressSet)
+			if err := egress.ensurePeerAddressSet(); err != nil {
+				klog.Errorf(err.Error())
+				continue
+			}
 		}
 
 		for _, toJSON := range egressJSON.To {
 			// Add IPBlock to egress network policy
 			if toJSON.IPBlock != nil {
 				egress.addIPBlock(toJSON.IPBlock)
 			}
-		}
 
-		localPodAddACL(np, egress)
-
-		for _, toJSON := range egressJSON.To {
-			egressPolicyHandler := policyHandler{
+			policyHandlers = append(policyHandlers, policyHandler{
 				gress:             egress,
 				namespaceSelector: toJSON.NamespaceSelector,
 				podSelector:       toJSON.PodSelector,
-				addrSet:           hashedLocalAddressSet,
-				peerMap:           peerPodAddressMap,
-			}
-			policyHandlers = append(policyHandlers, egressPolicyHandler)
+			})
 		}
+		localPodAddACL(np, egress)
 		np.egressPolicies = append(np.egressPolicies, egress)
 	}
 	np.Unlock()
+
```

Comment:
delete extra line

Reply:
code is a bit different now, and I think it justifies the newline here.